File: java-concepts/src/main/java/org/demo/expressions/Expressions.java
package org.demo.expressions;

public class Expressions {
    public void arithmeticExpressions() {
        int a = 1;
        int b = 2;

        System.out.println("Increment: " + (a++));
        System.out.println("Decrement: " + (a--));
        System.out.println("Plus: " + (a + b));
        System.out.println("Minus: " + (a - b));
        System.out.println("Multiplication: " + (a * b));
        System.out.println("Division: " + ((float) a / b));
        System.out.println("Remainder (Modulo): " + (-5.2f % 2.0f));
    }

    public void logicalExpressions() {
        boolean a = true;
        boolean b = false;

        System.out.println("Negation: " + (!a));
        System.out.println("AND: " + (a && b));
        System.out.println("OR: " + (a || b));
        System.out.println("XOR: " + (a ^ b));
        System.out.println("Negation: " + (!a));
    }

    public void orderExpressions() {
        int a = 1;
        int b = 2;

        System.out.println("Less Than: " + (a < b));
        System.out.println("Less Than or Equal: " + (a <= b));
        System.out.println("Greater Than: " + (a > b));
        System.out.println("Greater Than or Equal: " + (a >= b));
    }

    public void equalityExpressions() {
        int valueType1 = 1;
        int valueType2 = 1;

        System.out.println("Equal: " + (valueType1 == valueType2));
        System.out.println("Not Equal: " + (valueType1 != valueType2));

        String referenceType1 = "1";
        String referenceType2 = "1";

        System.out.println("Equal: " + (referenceType1.equals(referenceType2)));
        System.out.println("Not Equal: " + (!referenceType1.equals(referenceType2)));
        System.out.println("Equal memory reference: " + (referenceType1 == referenceType2));
    }
    
    public void ternaryOperator() {
        boolean condition = true;
        String result = condition ? "True value" : "False value";
    }
}

File: csharp-concepts/expressions/Expressions.cs
namespace csharp_concepts.expressions;

public class Expressions
{
    public void ArithmeticExpressions()
    {
        var a = 1;
        var b = 2;

        Console.WriteLine($"Increment: {a++}");
        Console.WriteLine($"Decrement: {a--}");
        Console.WriteLine($"Plus:{a + b}");
        Console.WriteLine($"Minus:{a - b}");
        Console.WriteLine($"Multiplication:{a * b}");
        Console.WriteLine($"Division:{a / b}");
        Console.WriteLine($"Remainder (Modulo):{-5.2 % 2.0}");
    }

    public void LogicalExpressions()
    {
        var a = true;
        var b = false;

        Console.WriteLine($"Negation:{!a}");
        Console.WriteLine($"AND:{a && b}");
        Console.WriteLine($"OR:{a || b}");
        Console.WriteLine($"XOR:{a ^ b}");
    }

    public void OrderExpressions()
    {
        var a = 1;
        var b = 2;

        Console.WriteLine($"Less Than:{a < b}");
        Console.WriteLine($"Less Than or Equal:{a <= b}");
        Console.WriteLine($"Greater Than:{a > b}");
        Console.WriteLine($"Greater Than or Equal:{a >= b}");
    }

    public void EqualityExpressions()
    {
        var valueType1 = 1;
        var valueType2 = 1;

        Console.WriteLine($"Equal:{valueType1 == valueType2}");
        Console.WriteLine($"Not Equal:{valueType1 != valueType2}");

        var referenceType1 = new Expressions();
        var referenceType2 = new Expressions();

        Console.WriteLine($"Equal memory reference: {referenceType1 == referenceType2}");
    }

    public void TernaryOperator()
    {
        var condition = true;
        var result = condition ? "True value" : "False value";
    }
}